test/framework/docker_logcollector.go

```diff
@@ -76,6 +76,25 @@ func (k DockerLogCollector) CollectMachinePoolLog(ctx context.Context, _ client.
 	return kerrors.NewAggregate(errs)
 }
 
+func (k DockerLogCollector) CollectInfrastructureLogs(ctx context.Context, clusterName string, outputPath string) error {
+	containerRuntime, err := container.NewDockerClient()
+	if err != nil {
+		return err
+	}
+	ctx = container.RuntimeInto(ctx, containerRuntime)
+
+	lbContainerName := clusterName + "-lb"
+
+	f, err := fileOnHost(filepath.Join(outputPath, fmt.Sprintf("%s.log", lbContainerName)))
+	if err != nil {
+		return err
+	}
+
+	defer f.Close()
+
+	return containerRuntime.ContainerDebugInfo(ctx, lbContainerName, f)
```

Comment:
Could we also pull the haproxy.cfg? Might be useful to see what state it's in after shutdown time.

Reply:
Yes should be possible but not in all cases, e.g. only for "still running" lb containers, not for Exited ones.

Reply:
Yeah - probably not especially useful for this use case.

Reply:
Also, pulling down the haproxy config file seems wrong to me for this function. Maybe we can do that in a separate PR :-)